presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -169,33 +178,25 @@ public Page getNextPage()
 
             Block[] blocks = new Block[hiveColumnIndexes.length];
             for (int fieldId = 0; fieldId < blocks.length; fieldId++) {
-                Type type = types.get(fieldId);
                 if (constantBlocks[fieldId] != null) {
                     blocks[fieldId] = constantBlocks[fieldId].getRegion(0, batchSize);
                 }
                 else {
-                    int fieldIndex;
-                    if (useParquetColumnNames) {
-                        fieldIndex = fileSchema.getFieldIndex(columnNames.get(fieldId));
+                    Type type = types.get(fieldId);
+                    if (ROW.equals(type.getTypeSignature().getBase())) {
+                        blocks[fieldId] = buildRowBlock(type.getTypeSignature().getParameters(), batchSize, 1);
```

Comment:
the constant `1` is confusing, you can create a recursive helper method to pass that to `buildRowBlock`.